examples/interface/index.test.js

```diff
@@ -28,6 +28,31 @@ query {
   done();
 });
 
+it('User.field1 with typename', async (done) => {
+  await make.User({
+    _id: 'the-id',
+    type: 'admin',
+    mongoA: 123,
+  });
+  const result = await run(gql, `
+query {
+  user(id: "the-id") {
+    __typename
+    field1
+  }
+}
+`);
+  expect(result).toEqual({
+    data: {
+      user: {
+        __typename: 'AdminUser',
+        field1: 123,
+      },
+    },
+  });
+  done();
+});
+
```

Comment:
@b1f6c1c4 Actually I'm not sure this test would failed without the PR, because I had not error, just an undefined projection. 🤔

Reply:
That's because errors are caught [here](https://github.com/b1f6c1c4/graphql-advanced-projection/blob/master/src/projection.js#L77-L86). It makes testing useless, unfortunately. Maybe I should consider adding a config like `{ error: 'ignore' | 'bail' }` 😅